plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/toDataFrame.kt

```diff
@@ -190,34 +190,56 @@ class Exclude1 : AbstractInterpreter<Unit>() {
         dsl.excludeProperties.addAll(properties.arguments.filterIsInstance<FirCallableReferenceAccess>())
     }
 }
+
 @Suppress("INVISIBLE_MEMBER")
 @OptIn(SymbolInternals::class)
 internal fun KotlinTypeFacade.toDataFrame(
     maxDepth: Int,
     arg: ConeTypeProjection,
     traverseConfiguration: TraverseConfiguration,
 ): PluginDataFrameSchema {
+
     fun ConeKotlinType.isValueType() =
         this.isArrayTypeOrNullableArrayType ||
             this.classId == StandardClassIds.Any ||
             this.classId == StandardClassIds.String ||
             this.classId == StandardClassIds.Boolean ||
-            classId in setOf(Names.DURATION_CLASS_ID, Names. LOCAL_DATE_CLASS_ID, Names.LOCAL_DATE_TIME_CLASS_ID, Names.INSTANT_CLASS_ID) ||
+            classId in setOf(
+            Names.DURATION_CLASS_ID,
+            Names.LOCAL_DATE_CLASS_ID,
+            Names.LOCAL_DATE_TIME_CLASS_ID,
+            Names.INSTANT_CLASS_ID,
+            Names.DATE_TIME_PERIOD_CLASS_ID,
+            Names.DATE_TIME_UNIT_CLASS_ID,
+            Names.TIME_ZONE_CLASS_ID
+        ) ||
             this.isSubtypeOf(session.builtinTypes.numberType.type, session) ||
-            this.isSubtypeOf(StandardClassIds.Number.constructClassLikeType(emptyArray(), isNullable = true), session) ||
+            this.isSubtypeOf(
+                StandardClassIds.Number.constructClassLikeType(emptyArray(), isNullable = true),
+                session
+            ) ||
             this.toRegularClassSymbol(session)?.isEnumClass ?: false ||
             this.isSubtypeOf(
                 ConeClassLikeTypeImpl(
                     ConeClassLikeLookupTagImpl(
-                        ClassId(FqName("java.time.temporal"), Name.identifier("Temporal"))
+                        ClassId(FqName("java.time.temporal"), Name.identifier("TemporalAccessor"))
+                    ), arrayOf(), isNullable = false
+                ), session
+            ) ||
+            this.isSubtypeOf(
+                ConeClassLikeTypeImpl(
```

Comment:
let's refactor it a little, extract class ids in Names object and then call Names.TEMPORAL.constructClassLikeType instead of ConeClassLikeTypeImpl...

Reply:
Ok, btw why `isNullable=false` here?

Reply:
Good catch! For number type i have two subtypeOf, one with nullable type and other with non-nullable. But 1 check with nullable = true will be enough and it's more correct. Test case that works incorrectly now:
```
class Test(val temporal: Temporal?)

listOf(Test(LocalDateTime.parse("01/01/08 00:02:04", DateTimeFormatter.ofPattern("MM/dd/yy HH:mm:ss"))), Test(null)).toDataFrame(maxDepth = 2)   
```